ssh-notes/ssh-connection-problems.md

```diff
@@ -0,0 +1,138 @@
+# Connection Problems
+
+This is intended to address a very specific problem you may encounter when
+
+- trying to use SSH
+- from a WSL 2 distrubution
+- running under Windows
+
+## How to Tell if you have the problem
+
+The most obvious symptom is that when you try and perform any SSH related activities, the SSH program appears to hang.
+
+This problem will occur with any SSH command that has to try and send a key across the network, so it could affect connections to any SSH server (e.g. Azure).
+
+To check if you are having the specific connection problem this article attempts to address, do the following:
+
+1. Start a WSL shell
+2. Run the following command
+  ```bash
+  ssh -T -vv git@github.com
+  ```
+3. If you are suffering from the connection issue you will see the following output:
+
+  ```text
+  openssH 8.2p1 Ubuntu-4, openssL 1.1.1f 31 mar 2828
+  debug1: Reading configuration data /etc/ssh/ssh_config
+  debug2: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files
+  debug1: /etc/ssh/ssh_config line 21: Applying options for *
+  debugl: Connecting to github.ccm [148.82.121.3] port 22.
+  debugl: Connection established.
+  ...
+  ... Lots of messages here ...
+  ...
+  debugl: Authenticating to github.ccn:22 as 'git'
+  debugl: . KEXINIT sent
+  debugl: . SSH2 KEXINIT received
+  debugl: kex: algoritfun: curve25519-sha256
+  debugl: kex: host key algoritfun: ecdsa-sha2-nistp256
+  debugl: kex: server->client cipher: chacha2ø-p01y13øS@penssh.ccm (implicit> compression:
+  debugl: kex: client->server cipher: chacha2ø-p01y13øS@penssh.ccm 'u: (implicit> compression:
+  debugl: expecting SSH2_KEX_ECDH_REPLY
+  ```
+
+## How to Fix the Problem - Preparation
+
+The problem is caused by a mismatch in the configuration of the network cards in Windows and the 'virtual' network card in the WSL machine.  As preparationto fix the problem you first need to do the following:
+
+1. Start a Windows command prompt, and run the following command.
+   ```text
+   netsh interface ipv4 show interfaces
+   ```
```

Comment:
```suggestion

   ```text
   netsh interface ipv4 show interfaces
   ```

```

Reason: MD031/blanks-around-fences: Fenced code blocks should be surrounded by blank lines (markdownlint)